applications/asset_tracker_v2/tests/gps_module/src/gps_module_test.c

```diff
@@ -0,0 +1,216 @@
+	__wrap_gps_init_AddCallback(&gps_init_callback);
+	__wrap_module_start_ExpectAndReturn(&expected_module_data, 0);
+
+	bool ret =
+		GPS_MODULE_EVT_HANDLER((struct event_header *)app_module_event);
+	TEST_ASSERT_EQUAL(0, ret);
+
+	free(app_module_event);
+}
+
+static void setup_gps_module_in_running_state(void)
+{
+	setup_gps_module_in_init_state();
+
+	struct data_module_event *data_module_event = new_data_module_event();
+
+	data_module_event->type = DATA_EVT_CONFIG_INIT;
+	data_module_event->data.cfg.gps_timeout = 60;
+
+	bool ret = GPS_MODULE_EVT_HANDLER(
+		(struct event_header *)data_module_event);
+	TEST_ASSERT_EQUAL(0, ret);
+
+	free(data_module_event);
+}
+
+static struct gps_event *generate_gps_fix_evt(void)
+{
+	static struct gps_event evt = {
+		.type = GPS_EVT_NMEA_FIX,
+	};
+
+	evt.nmea.len = snprintf(evt.nmea.buf, GPS_NMEA_SENTENCE_MAX_LENGTH,
+				"%s", GPS_NMEA_SAMPLE);
+
+	return &evt;
+}
+
+/* Test whether sending a APP_EVT_DATA_GET event to the GPS module generates
+ * the GPS_EVT_ACTIVE event, when the GPS module is in the running state.
+ */
+void test_gps_start(void)
+{
+	/* Pre-condition. */
+	setup_gps_module_in_running_state();
+
+	/* Setup expectations.  */
+	__wrap_gps_start_ExpectAnyArgsAndReturn(0);
+
+	__wrap__event_submit_Stub(&validate_gps_active_evt);
+	/* Stimulus. */
+	struct app_module_event *app_module_event = new_app_module_event();
+
+	app_module_event->type = APP_EVT_DATA_GET;
+	app_module_event->count = 1;
+	app_module_event->data_list[0] = APP_DATA_GNSS;
+
+	bool ret =
+		GPS_MODULE_EVT_HANDLER((struct event_header *)app_module_event);
+	TEST_ASSERT_EQUAL(0, ret);
+
+	/* Cleanup */
+	free(app_module_event);
+	__wrap__event_submit_Stub(NULL);
+}
+
+/* Test whether sending a APP_EVT_DATA_GET event to the GPS module generates
+ * the GPS_EVT_ACTIVE event, when the GPS module is in the running state.
+ */
+void test_gps_fix(void)
+{
+	resetTest();
+	/* Pre-condition. */
+	setup_gps_module_in_running_state();
+
+	/* Make gps_module start gps driver. */
+	__wrap_gps_start_ExpectAnyArgsAndReturn(0);
+	__wrap__event_submit_ExpectAnyArgs();
+
+	struct app_module_event *app_module_event = new_app_module_event();
+
+	app_module_event->type = APP_EVT_DATA_GET;
+	app_module_event->count = 1;
+	app_module_event->data_list[0] = APP_DATA_GNSS;
+
+	bool ret =
+		GPS_MODULE_EVT_HANDLER((struct event_header *)app_module_event);
+	TEST_ASSERT_EQUAL(0, ret);
+
+	free(app_module_event);
+
+	/* Simulate a GPS FIX and send it to the GPS module.  */
+	struct gps_event *gps_fix_evt = generate_gps_fix_evt();
+
+	/* Expect the gps module to generate an event with the fix data. */
+	__wrap__event_submit_Stub(&validate_gps_data_ready_evt);
+
+	strncpy(exp_nmea, gps_fix_evt->nmea.buf, sizeof(exp_nmea) - 1);
+	exp_nmea[sizeof(exp_nmea) - 1] = '\0';
+
+	gps_module_gps_evt_handler(NULL, gps_fix_evt);
+	__wrap__event_submit_Stub(NULL);
+}
+
+/* It is required to be added to each test. That is because unity is using
```

Comment:
```suggestion
/* This is required to be added to each test because unity is using
```

Reply:
I have changed this as follows and moved it to the top of the file along with other declarations. 

`/* The following is required because unity is using different main signature
 * (returns int) and zephyr expects main which does not return value.
 */
extern int unity_main(void);`